receiver/dcgmreceiver/client.go

```diff
@@ -367,13 +369,13 @@ func (client *dcgmClient) getDeviceMetrics() map[uint]deviceMetrics {
 	return out
 }
 
-func (client *dcgmClient) issueWarningForFailedQueryUptoThreshold(dcgmName string, reason string) {
+func (client *dcgmClient) issueWarningForFailedQueryUptoThreshold(dcgmName string, limit int, reason string) {
 	client.deviceMetricToFailedQueryCount[dcgmName]++
 
 	failedCount := client.deviceMetricToFailedQueryCount[dcgmName]
-	if failedCount <= maxWarningsForFailedDeviceMetricQuery {
-		client.logger.Warnf("Unable to query '%s' on '%s'", dcgmName, reason)
-		if failedCount == maxWarningsForFailedDeviceMetricQuery {
+	if failedCount <= limit {
+		client.logger.Warnf("%s", reason)
```

Comment:
`logger.Warnf("%s", reason)` is the same as `logger.Warn(reason)` (https://pkg.go.dev/go.uber.org/zap#SugaredLogger.Warn). But I can fix that after we merge this PR…